spec/Section 3 -- Type System.md

```diff
@@ -211,14 +211,19 @@ type MyMutationRootType {
 
 **Default Root Operation Type Names**
 
-While any type can be the root operation type for a GraphQL operation, the type
-system definition language can omit the schema definition when the {`query`},
-{`mutation`}, and {`subscription`} root types are named {"Query"}, {"Mutation"},
-and {"Subscription"} respectively.
+:: While any type can be the root operation type for a GraphQL operation, the
+_default root operation type name_ for the {`query`}, {`mutation`}, and
+{`subscription`} root types are {"Query"}, {"Mutation"}, and {"Subscription"}
+respectively.
+
+The type system definition language can omit the schema definition when each
+root type present uses its respective _default root operation type name_ and no
+other type uses a default root operation type name.
 
 Likewise, when representing a GraphQL schema using the type system definition
-language, a schema definition should be omitted if it only uses the default root
-operation type names.
+language, a schema definition should be omitted if each root type present uses
```

Comment:
Given a schema with the mutation root type with name CustomMutationRootType, but query root type query, presumably when printing the schema (converting to string), the schema definition should only include the entry for overriding the default mutation type name and not respecify the default query root type name.

Assuming that is true, perhaps we need a sentence within this expanded section that says so explicitly.

Reply:
> The type system definition language can omit ... 

That does not seem right. The 'language' omits something? The SDL file maybe?

Reply:
@yaacovCR 

> Given a schema with the mutation root type with name CustomMutationRootType, but query root type query, presumably when printing the schema (converting to string), the schema definition should only include the entry for overriding the default mutation type name and not respecify the default query root type name.

No; the schema definition is either omitted or included, it's never included partially. Otherwise my Virus example above would be construed to support mutation operations (it does not).

Reply:
@rivantsov That text is unchanged from the current spec, that is a topic for a separate PR.

Reply:
> No; the schema definition is either omitted or included, it's never included partially. Otherwise my Virus example above would be construed to support mutation operations (it does not).

@benjie i guess what I am asking is whether the schema definition should be only partially included __when it can __ — ie my example rather than yours.

I imagine this is in line with the idea of skipping it entirely when possible. I am not sure this is desirable, but it would work.

Reply:
I don't think that's desirable, and either way is outside the scope of this PR.